src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -829,107 +832,152 @@ subroutine init_atm_static(mesh, dims, configs)
+                        !
+                        ! This field only matters for land cells, and for all but the outermost boundary cells,
+                        ! we can safely assume that the nearest model grid cell contains the pixel (else, a different
+                        ! cell would be nearest)
+                        !
+                        ! Since values are currently not scaled, we can compare missing_value, which also
+                        ! isn't scaled, we can compare the two without scaling.
+                        if (landmask(res % id) == 1 .and. i8val /= int(missing_value, kind=I8KIND)) then
+                            snoalb_integer(res % id) = snoalb_integer(res % id) + i8val
+                            nhs(res % id) = nhs(res % id) + 1
+                        end if
+
+                        !
+                        ! When a pixel maps to a non-land cell or is not a missing value, the values are not accumulated
+                        ! aboved; however, these pixels may still reside in an own cell, in which case we will still need
+                        ! to push the tile's neighbors unto the stack for processing.
+                        !
+                        if (res % id <= nCellsSolve) then
+                            all_pixels_mapped_to_halo_cells = .false.
+                        end if
+                    ! For outermost land cells, additional work is needed to verify that the pixel
+                    ! actually lies within the nearest cell
+                    else
+                        if (mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
+                                          nEdgesOnCell(res % id), verticesOnCell(:,res % id), xVertex, yVertex, zVertex)) then
+
+                            ! Since values are currently not scaled, we can compare missing_value, which also
+                            ! isn't scaled, we can compare the two without scaling.
+                            if (landmask(res % id) == 1 .and. i8val /= int(missing_value, kind=I8KIND)) then
+                                snoalb_integer(res % id) = snoalb_integer(res % id) + i8val
+                                nhs(res % id) = nhs(res % id) + 1
+                            end if
+
+                            !
+                            ! When a pixel maps to a non-land cell or is not a missing value, the values are not accumulated
+                            ! aboved; however, these pixels may still reside in an own cell, in which case we will still need
+                            ! to push the tile's neighbors unto the stack for processing.
+                            !
+                            if (res % id <= nCellsSolve) then
+                                all_pixels_mapped_to_halo_cells = .false.
+                            end if
+                        end if
+                    end if
```

Comment:
Just for consideration, there's an alternative for the code between lines 899 and 942 that doesn't require any duplication of code and that doesn't require any nested if-tests:
```fortran
                    !
                    ! If the nearest cell center is that of an outer-boundary cell (i.e., bdyMaskCell == nBdyLayers),
                    ! we need to check whether the pixel is actually contained in that cell, since the cell center
                    ! might simply have been the closest point to a pixel that lies outside of a limited-area domain
                    !
                    if (bdyMaskCell(res % id) == nBdyLayers .and. &
                        .not. mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
                                           nEdgesOnCell(res % id), verticesOnCell(:,res % id), xVertex, yVertex, zVertex)) then

                           ! No further processing for this pixel is needed...
                           cycle

                    end if

                    !
                    ! This field only matters for land cells. If the cell containing this pixel is a land cell,
                    ! and if the pixel isn't a missing value, accumulate the pixel value in the cell.
                    ! NB: The i8val value is not scaled by scale_factor and neither is missing_value, so no
                    !     scaling is required in the comparison, below.
                    !
                    if (landmask(res % id) == 1 .and. i8val /= int(missing_value, kind=I8KIND)) then
                        snoalb_integer(res % id) = snoalb_integer(res % id) + i8val
                        nhs(res % id) = nhs(res % id) + 1
                    end if

                    !
                    ! Regardless of whether we accumulate a pixel value, we need to consider whether we have pixels
                    ! from a tile that map to owned (i.e., non-halo) cells
                    !
                    if (res % id <= nCellsSolve) then
                        all_pixels_mapped_to_halo_cells = .false.
                    end if
```
@MiCurry What do you think of this alternative?

Reply:
@mgduda I like this a lot better. I'll implement it for this PR.

Reply:
Actually, thinking a bit more about this change, I’d like to forgo it for now. I think I had mentioned earlier that I would like to remove the duplicated code, but I figured it would be best if we did it when we wrote a unified function. I’d also like to keep this code as more or less consistence with the other interpolated fields, knowing that it (and the other interpolations) will be updated in a future PR.

Reply:
That's fine. In this case, there are a few minor mistakes in the comments (which would have been replaced by the proposed code and comments) that we should fix before merging this PR. I'll leave comments for these.